src/app/routes/assets/verification-method/verification-method.component.ts

```diff
@@ -22,8 +22,15 @@ export class VerificationMethodComponent implements OnInit {
   verificationsAmount: number;
   dataSource: PublicKey[] = [];
   selectControl = new FormControl('', [Validators.required]);
+
+  publicKeyExistValidator(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
```

Comment:
```suggestion
      listContainsValidator(source: unknown[], property: string): ValidatorFn {
    return (control: AbstractControl): ValidationErrors | null => {
      const exists = source.find(key => key[property] === control.value);
      return exists ? {listContains: {value: control.value}} : null;
    };
  }
```
What do you think? This way we can remove this validator from component, writing unit tests will be pretty easy.
The only thing is that you need to set validators later like this: `this.publicKey.setValidators([Validators.required, isHexValidator, this.publicKeyExistValidator(this.dataSource, 'publicKeyHex')]);`, not when initializing. For example in handleLoadedPublicKeys.

@VOSpdUkraine please add your thoughts on this.

Reply:
Yeah. I was thinking of putting the function in utils/validators folder. But I didnt see any other component using it. So I left it there

Reply:
Ok, it would be good to have tests for it imo.
I think having it as a function will make testing a lot easier. -> you do not have to mock component properties. However it's your call, both solutions are fine for me.